File: ArraysAndHashing/169.cpp
/*
 
 	Majority element
	
	Input: [2, 2, 1, 1, 1, 3, 3]
	
	Output: 1
 
 */


#include <iostream>
#include <unordered_map>
using namespace std;


int MajorityElement(vector<int>& nums){
	unordered_map<int, int> tmp;
	for(int n: nums){
		tmp[n]++;
	}
	
	int MajorityKey = nums[0];
	int MaxCount = 0;
	for(auto& p: tmp){
		if(p.second > MaxCount){
			MaxCount = p.second;
			MajorityKey = p.first;
		}
	}
	return MajorityKey;

}

int main(){
	vector<int> nums = {2, 2, 1, 1, 1, 3, 3, 3, 3, 1, 1};
	auto result = MajorityElement(nums);
	
	cout << result << endl;
	return 0;
}

File: ArraysAndHashing/py/169.py
"""
	Majority element
	
	Input: [2, 2, 1, 1, 1, 3, 3]
	
	Output: 1

"""



def MajorityElement(nums: list[int]) -> int:
	tmp = {}
	for i in nums:
		if i in tmp:
			tmp[i] += 1
		else:
			tmp[i] = 1
	max = (0, 0)
	for num, count in tmp.items():
		if max[1] < count:
			max = (num, count)
	return max[0]
	
	
	
nums = [2, 2, 1, 1, 3, 3, 4, 4, 4, 4]
result = MajorityElement(nums)

print(result)